### ISSUE TITLE
After host recover from reconnecting stack's don't update in that same host

### ISSUE THREAD

**Issue by fernandoneto** [rancher/rancher#3175]
After a host recover from a reconnecting state the stack's in that host don't update automatically. I have to update the stack again to changes take effect in the affected host.

**Comment by cusspvz** [comment]
@fernandoneto is referring to global services, whenever you get a host online or add a new host, it should schedule a deployment of a container into those, thats not occurring. (it was on older versions)

Reproducible steps:
* Create an empty environment
* Create a stack
* Add a global service
* Add a host
* At this time, service should be scheduled on the recently added host, it won't be unless you edit the service and save it.

**Comment by deniseschannon** [comment]
I am unable to reproduce. If I add a host, the service is schedule on the recently added host.

When you "edit" the service, what are you doing? Are you just clicking on edit and then save?

What version of Rancher are you using? Can you share your docker-compose.yml of the global service.

**Comment by cusspvz** [comment]
Has happened with a host that went from `Reconnecting` state to `Active`, and with a host we added back up.

The services affected with this behavior were: a load-balancer and one maintenance image, both are running as global services.
It initially happened with me more than once, always with `Reconnecting` hosts being back up (instances were being deleted but not populated). Seems this behavior is intermittent because today it didn't happened once.

```
Component	Version
Rancher	v0.51.0
Cattle	v0.130.0
User Interface	v0.78.0
Rancher Compose	v0.7.0
```

```yaml
autotune:
  environment:
    net__ipv4__tcp_max_orphans: '40000'
    net__ipv4__tcp_fin_timeout: '30'
    docker_remove_containers: '0'
    docker_remove_images: '1'
  labels:
    io.rancher.scheduler.global: 'true'
  tty: true
  image: cusspvz/autotune:1.1.1
  privileged: true
  volumes:
  - /var/run/docker.sock:/var/run/docker.sock
  stdin_open: true
  net: host
```

**Comment by deniseschannon** [comment]
To be clear, you had a host in reconnecting state, while it was in reconnecting state, you added a global service to Rancher. The reconnecting host became "Active", but the service never got deployed onto this host (Even though it should have?)?

This will be a bit harder for me to test out, but let me first get the use case correct.

**Comment by cusspvz** [comment]
> while it was in reconnecting state, you added a global service to Rancher

Nope, the service was already deployed on that host, it went into Reconnecting state, all the global services were destroyed (main load balancers, maintenance instances and others we have as global for haendpoints such as STUN services). After it went up, it didn't schedule none of them.

Same was reproducible with a host we've added into the environment, but that one didn't had any services on it.

**Comment by deniseschannon** [comment]
This is strange because I definitely don't have any issues with global services and having them schedule onto hosts that were just introduced into the environment. 

Are you saying that in this environment, when you add any host, none of the global services are being scheduled onto it?

**Comment by cusspvz** [comment]
> Are you saying that in this environment, when you add any host, none of the global services are being scheduled onto it?

Only after we edit and save those services.

We have migrated our rancher-server into another machine, and since then I didn't catch (til yet) any reconnecting host (because we are restarting the server every 2 hours, which solves temporarily our issue, we had to do so because our endpoint hosts were having their main load balancers removed once they went into reconnecting mode). As soon as we got a similar situation I could screencast it and share it here.

Little bit off-topic, I've seen that @ibuildthecloud is working on a new network agent and has already built-in on latest *rc*s. Do you know when the next minor release will be available?

**Comment by cusspvz** [comment]
@deniseschannon I've experienced the same a few minutes ago but unfortunately didn't screencasted.

But I have more info, I think it might be related with issue #2196 and last @alena1108 fix regarding the removal of instances when a host goes down, because the global services that weren't deployed were the ones removed while the host was on Reconnecting state.

**Comment by fernandoneto** [comment]
It seems that it is working properly in version v0.59.1 . I just add a new host to the cluster and services marked as global start automatically.

**Comment by deniseschannon** [comment]
@fernandoneto @cusspvz I think this is related to https://github.com/rancher/rancher/issues/3480, but I don't see it fixed on v0.59.1

Since it seems to be working for your issues, could i close in favor of #3480?

**Comment by cusspvz** [comment]
@deniseschannon: Nice catch, seems that our "reconnecting hosts" could be the culprit on this.
Yes, you can close this.

So far I've seen, we are always experiencing the deepest issues, the "reconnecting hosts" one is still unsolved, our work around is a cron job restarting `rancher-server` each 2 hours, which has already demonstrated to cause some side effects. x) On the other hand, we can now deploy faster. <3 rancher & docker

I'm a little bit out of rancher's management by now, @fernandoneto is mostly more involved and on charge of it, but feel free to include me on and mention me!
